lms/events/event.py

```diff
@@ -101,3 +102,33 @@ def _get_assignment_id(self):
             return assignment.id
 
         return None
+
+
+class AdminChangeEvent(BaseEvent):
+    def __init__(self, instance, *args, **kwargs):
+        self._instance = instance
+        super().__init__(*args, **kwargs)
+
+    def _get_data(self):
+        changes = {}
+        instance_details = inspect(self._instance)
+        for attr in instance_details.attrs:
+            history = instance_details.get_history(attr.key, True)
+
+            if not history.has_changes():
+                continue
+
+            changes[attr.key] = (1, history.added[0])
+
+        return {"user": self.request.identity.userid, "changes": changes}
+
+    @staticmethod
+    def notify_changes(request, instance):
+        if request.db.is_modified(instance):
+            request.registry.notify(
+                AdminChangeEvent(
+                    instance=instance,
+                    request=request,
+                    type=AdminChangeEvent.Type.ADMIN_PAGES_UPDATE,
+                )
+            )
```

Comment:
I think it's worth thinking about whether we can work this out another way. I like being able to do:

```python
request.registry.notify(
    AdminChangeEvent(
        instance=instance,
        request=request,
        type=AdminChangeEvent.Type.ADMIN_PAGES_UPDATE,
    )
)
```

In the code without having to worry about it

Reply:
This allows to `notify` without having to worry if there are changes or not (when we don't want a new row in the DB.

Later it can be extended to handle new objects, deletions etc with the same `notify(object)`

Reply:
I was thinking we could have a "should_notify: bool" which could be set from `request.db.is_modified(instance)`. Then the subscriber would just "Nope" out if it's false.

But this is fine. There's no need to have a "system" yet (despite everything else we are doing!) as there's only one caller.